### PR TITLE
Fix bug 1124419 - add a waffle flag around profile creation

### PR DESCRIPTION
**Pull request by maurodoglio**

### CODE REVIEW

File: kuma/users/views.py [hunk 1]
```diff
@@ -405,7 +406,11 @@ def get_context_data(self, **kwargs):
                                               .filter(reduced_or_query))
         else:
             matching_accounts = SocialAccount.objects.none()
+
+        registration_disabled = waffle.flag_is_active(self.request,
```
Comment (jezdez) [inline kuma/users/views.py]: This is the wrong place to put it, use the `kuma.users.adapters.KumaSocialAccountAdapter.is_open_for_signup` method to specify it. It shold automatically render a "not enabled" page.

**Comment by jezdez** [comment]
This needs updates for tests and docs.

**Comment by groovecoder** [comment]
Thanks for the contribution @maurodoglio. :+1: @jezdez comments - `KumaSocialAccountAdapter` is a `django-allauth` class that lets us control the entire auth code.

Here's an example of a test (`test_custom_css_waffle`) that checks specifically for a waffle flag on a feature:

https://github.com/mozilla/kuma/blob/master/kuma/wiki/tests/test_templates.py#L60-73

**Comment by maurodoglio** [comment]
@jezdez @groovecoder yeah, my first attempt was quite naif :smile: thanks for the directions!

**Comment by jezdez** [comment]
@maurodoglio No worries, not naif at all :)

File: kuma/users/views.py (hunk 1 above)
Comment (darkwing) [inline kuma/users/views.py]: I don't think this is even necessary, right?  We can simply do the waffle check within the template?

**Comment by maurodoglio** [comment]
Sorry it took so long :sweat:
@jezdez  is there a specific place in the docs where you would like me to describe this feature?

**Comment by groovecoder** [comment]
Probably in the "Enable Important Site Features" part of the docs. ([src](https://github.com/mozilla/kuma/blob/master/docs/installation-vagrant.rst#enable-important-site-features), [live](http://kuma.readthedocs.org/en/latest/installation-vagrant.html#enable-important-site-features))

**Comment by groovecoder** [comment]
I'm going to pick this up. I'll make a new doc for feature flips and put an inventory of waffle flags and constance configs in it.

**Comment by groovecoder** [comment]
Superseded by https://github.com/mozilla/kuma/pull/3091 with a rebase and docs.

**Pull request closed without merging**